src/Sentry/PrefixOrRegexPattern.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+using Sentry.Internal;
+
+namespace Sentry;
+
+/// <summary>
+/// Provides a pattern that can be used to match against namespaces as either a prefix or regular expression.
+/// </summary>
+internal class PrefixOrRegexPattern
```

Comment:
Any reason this needs to be a single class that handles both cases? How about an interface (e.g. `StringFilter`) with `IsMatch()` (one variant with separator arg and another one without it to really clean this up). Each implementation can just have its own code for prefix check/regex check (and substring check). Then we can leave the construction of the right implementation outside

Reply:
Frustratingly, this is actually a really good suggestion 😝 We do eventually have a bunch of other issues to close but I think this is a worthwhile refactoring to make. It should make the code a bit simpler and more composable. It should allows us to get rid of the `SubstringOrRegexPattern` class in the next major version bump, for example (deleting code 🎉 )

Reply:
OK done - thanks @vaind, I think it looks nicer now.